octodns_dnsmadeeasy/__init__.py

```diff
@@ -186,6 +187,12 @@ class DnsMadeEasyProvider(BaseProvider):
             'TXT',
         )
     )
+    # Regex to replace any pair of double quotes that aren't escaped with a backslash. Used as a delimiter in long TXT
+    # records.
+    #
+    #  Will match: Alpha""Bravo
+    #  Will not match: Alpha\""Bravo
+    TXT_RECORD_VALUE_DELIMITER_PATTERN = re.compile(r'(?<!\\)\"\"')
```

Comment:
The regex appears to match both of the examples above:

```python
>>> import re
>>> TXT_RECORD_VALUE_DELIMITER_PATTERN = re.compile(r'(?<!\\)\"\"')
>>> TXT_RECORD_VALUE_DELIMITER_PATTERN.sub('xx', 'Alpha""Bravo')
'AlphaxxBravo'
>>> TXT_RECORD_VALUE_DELIMITER_PATTERN.sub('xx', 'Alpha\""Bravo')
'AlphaxxBravo'
>>> TXT_RECORD_VALUE_DELIMITER_PATTERN.sub('xx', 'Alpha Bravo')
'Alpha Bravo'
```

Reply:
In the second test, your input string is already having the first double quote escaped. Telling python to treat the input as a raw string, I get:

```python
>>> import re
>>> TXT_RECORD_VALUE_DELIMITER_PATTERN = re.compile(r'(?<!\\)\"\"')
>>> print(TXT_RECORD_VALUE_DELIMITER_PATTERN.sub('xx', r'Alpha""Bravo'))
AlphaxxBravo
>>> print(TXT_RECORD_VALUE_DELIMITER_PATTERN.sub('xx', r'Alpha\""Bravo'))
Alpha\""Bravo
>>> print(TXT_RECORD_VALUE_DELIMITER_PATTERN.sub('xx', r'Alpha Bravo'))
Alpha Bravo
```

Reply:
Specifically the string handling here, Python is treating it in your test as escaping the quotes:

```php
>>> print('Alpha\""Bravo')
Alpha""Bravo
>>> print(r'Alpha\""Bravo')
Alpha\""Bravo
```

So in your second example, the regex is correctly working on the input.

Reply:
Doh, you're right. Wasn't paying close enough attention and remembering that `'` escapes as well.